nemo_skills/inference/reward_model.py

```diff
@@ -0,0 +1,204 @@
+
+    # making sure output dir exists
+    Path(cfg.output_file).absolute().parent.mkdir(parents=True, exist_ok=True)
+
+    # we currently assume the dataset is small enough to be loaded into memory
+    data = []
+    with open(cfg.input_file, "rt", encoding="utf-8") as fin:
+        for line in fin:
+            data.append(json.loads(line))
+
+    # skipping based on the offset first
+    data = data[cfg.offset :]
+
+    starting_idx = 0
+    if cfg.skip_filled:
+        try:
+            with open(cfg.output_file, "rt", encoding="utf-8") as fin:
+                starting_idx = len(fin.readlines())
+        except FileNotFoundError:
+            LOG.warning(f"File `{cfg.output_file}` not found, starting from scratch")
+
+    # additionally, skipping whatever is pre-filled, assuming offset didn't change
+    data = data[starting_idx:]
+    if cfg.prompt_config is None:
+        # fetching from the default for corresponding dataset
+        dataset_module = importlib.import_module(f"nemo_skills.dataset.{cfg.dataset}")
+        cfg.prompt_config = dataset_module.PROMPT_CONFIG
+
+    prompt = get_prompt(cfg.prompt_config, cfg.prompt_template, examples_type=cfg.examples_type)
+    LOG.info("Prompt used: %s", prompt)
+
+    # need to account for anything that's prefilled
+    if 0 <= cfg.max_samples <= starting_idx:
+        cfg.max_samples = 0
+
+    if starting_idx < cfg.max_samples:
+        cfg.max_samples -= starting_idx
+
+    if cfg.max_samples < 0 or cfg.max_samples > len(data):
+        cfg.max_samples = len(data)
+
+    if len(data) == 0:  # we might not have any examples if skip_filled=True
+        return
+
+    LOG.info("Example prompt:\nData dictionary: %s\nPrompt: %s", data[0], prompt.fill(data[0]))
+
+    if cfg.dry_run:
+        return
+
+    # setting buffering=1 to force to dump the output after every line, so that we can see intermediate generations
+    with open(cfg.output_file, "at" if cfg.skip_filled else "wt", encoding="utf-8", buffering=1) as fout:
+        data_points = []
+        for idx, data_point in tqdm(enumerate(data), initial=starting_idx, total=len(data) + starting_idx):
+            if idx >= cfg.max_samples:
+                break
+            data_points.append(data_point)
+
+            if len(data_points) == cfg.batch_size or idx == cfg.max_samples - 1:
+                outputs = llm.score(
+                    prompts=[prompt.fill(dp, include_generation=True) for dp in data_points],
+                )
+
+                for output, original_data_point in zip(outputs, data_points):
+                    # to make it easier to follow up with evaluation and limit accidental errors, we are adding
+                    # all of the ground-truth data to the output file alongside the generated solutions
+                    output.update(original_data_point)
+                    fout.write(json.dumps(output) + "\n")
+                data_points = []
+
+
+error_recovery_params = '\n' + get_fields_docstring(
```

Comment:
Please remove from here and below. This is specific to code execution and not used in RM

Reply:
9df6bed52a84b03ee8f3c462b1d3dfa2a0a90d56